state-reconstruct-storage/src/types.rs

```diff
@@ -73,20 +73,35 @@ pub trait Proto {
     }
 }
 
+/// Version of snapshot influencing the format of data stored in GCS.
+#[derive(Clone, Default, Debug, Serialize_repr, Deserialize_repr)]
+#[repr(u16)]
+pub enum SnapshotVersion {
+    /// Initial snapshot version. Keys in storage logs are stored as `(address, key)` pairs.
+    Version0 = 0,
+    /// Snapshot version made compatible with L1 recovery. Differs from `Version0` by including
+    /// hashed keys in storage logs instead of `(address, key)` pairs.
+    #[default]
+    Version1 = 1,
+}
+
 #[derive(Clone, Default, Debug, Serialize, Deserialize)]
 pub struct SnapshotHeader {
-    pub l1_batch_number: L1BatchNumber,
-    pub miniblock_number: MiniblockNumber,
+    pub version: SnapshotVersion,
+    #[serde(rename = "l1BatchNumber")]
```

Comment:
there's also `#[serde(rename_all = "camelCase")]` on the struct declaration - would that work here?